src/Controller/BackendCsvImportController.php

```diff
@@ -0,0 +1,243 @@
+     * @param ContaoFrameworkInterface $framework
+     * @param RequestStack             $requestStack
+     * @param TokenStorageInterface    $tokenStorage
+     */
+    public function __construct(
+        Connection $connection,
+        FlashBagInterface $flashBag,
+        ContaoFrameworkInterface $framework,
+        RequestStack $requestStack,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->connection   = $connection;
+        $this->flashBag     = $flashBag;
+        $this->framework    = $framework;
+        $this->requestStack = $requestStack;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Generate the import view for list wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     * @throws ResponseException
+     */
+    public function importListWizard(DataContainer $dc)
+    {
+        $csvImport = $this->createImportInstance();
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            return array_merge($data, $row);
+        });
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'listitems', $GLOBALS['TL_LANG']['MSC']['lw_import'][0]);
+    }
+
+    /**
+     * Generate the import view for option wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     * @throws ResponseException
+     */
+    public function importOptionWizard(DataContainer $dc)
+    {
+        $csvImport = $this->createImportInstance();
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            $data[] = [
+                'value' => $row[0],
+                'label' => $row[1],
+            ];
+
+            return $data;
+        });
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'options', $GLOBALS['TL_LANG']['MSC']['ow_import'][0]);
+    }
+
+    /**
+     * Generate the import view for table wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws RedirectResponseException
+     * @throws ResponseException
+     */
+    public function importTableWizard(DataContainer $dc)
+    {
+        $csvImport = $this->createImportInstance();
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            $data[] = $row;
+
+            return $data;
+        });
+
+        $csvImport->setSeparators([
+            CsvImportUtil::SEPARATOR_COMMA,
+            CsvImportUtil::SEPARATOR_SEMICOLON,
+            CsvImportUtil::SEPARATOR_TABULATOR,
+        ]);
+
+        return $this->runDefaultRoutine($csvImport, $dc, 'tableitems', $GLOBALS['TL_LANG']['MSC']['tw_import'][0]);
+    }
+
+    /**
+     * Create the CSV import instance
+     *
+     * @return CsvImportUtil
+     */
+    private function createImportInstance()
+    {
+        return new CsvImportUtil(
+            $this->connection,
+            $this->flashBag,
```

Comment:
I feel its still better for unit tests and dependency injection. Otherwise you would need to rely on the mocking the request object with a flash bag, right? I did have the same thoughts with the session wich is also available from `$request->getSession()`